pgmpy/tests/test_metrics/test_metrics.py

```diff
@@ -212,3 +214,123 @@ def test_shd(self):
     def test_shd_unequal_graphs(self):
         with self.assertRaises(ValueError, msg="The graphs must have the same nodes."):
             SHD(self.dag_4, self.dag_5)
+
+
+class TestGraphicalSelfCompatibility(unittest.TestCase):
+    """
+    Tests for pgmpy.metrics.self_compatibility_graphical, verifying:
+      - Perfect agreement (mean SHD = 0)
+      - Systematic edge flips (mean SHD = 2)
+      - Proper forwarding of estimator kwargs
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        # A simple A→B→C ground truth
+        cls.full_dag = DiscreteBayesianNetwork([("A", "B"), ("B", "C")])
+        rng = np.random.RandomState(0)
+        cls.data = pd.DataFrame(
+            {
+                "A": rng.randint(2, size=100),
+                "B": rng.randint(2, size=100),
+                "C": rng.randint(2, size=100),
+            }
+        )
+
+    def test_perfect_compatibility(self):
```

Comment:
We will need to test this on a real algorithm and not just on a mocked function.